FormalConjectures/ErdosProblems/123.lean

```diff
@@ -0,0 +1,113 @@
+is the sum of distinct elements of `A` such that no element divides another.
+
+Reference: [ErLe96] Erdős, P. and Lewin, M., _$d$-complete sequences of integers_. Math. Comp. (1996).
+-/
+def IsDComplete (A : Set ℕ) : Prop :=
+  ∀ᶠ n in atTop, ∃ s : Finset ℕ,
+    (s : Set ℕ) ⊆ A ∧                    -- The summands come from A
+    IsAntichain (· ∣ ·) (s : Set ℕ) ∧   -- No summand divides another
+    s.sum id = n                         -- They sum to n
+
+/--
+The set of all natural numbers of the form $p^a q^b$ where $a, b ≥ 0$.
+This represents numbers whose prime factors are restricted to those of $p$ and $q$.
+-/
+def PowersOfTwo (p q : ℕ) : Set ℕ :=
+  {n : ℕ | ∃ a b : ℕ, n = p^a * q^b}
+
+/--
+The set of all natural numbers of the form $p^a q^b r^c$ where $a, b, c ≥ 0$.
+This represents numbers whose prime factors are restricted to those of $p$, $q$, and $r$.
+-/
+def PowersOfThree (p q r : ℕ) : Set ℕ :=
+  {n : ℕ | ∃ a b c : ℕ, n = p^a * q^b * r^c}
+
+/--
+Given three natural numbers `a`, `b`, `c`, this is the set of all natural numbers
+of the form $a^k b^l c^m$ where $k, l, m ≥ 0$.
+-/
+def powersOfThree (a b c : ℕ) : Set ℕ :=
+  {n | ∃ k l m : ℕ, n = a^k * b^l * c^m}
+
+/--
+Helper predicate for pairwise coprimality of three integers.
+-/
+def PairwiseCoprime (a b c : ℕ) : Prop :=
+  Nat.Coprime a b ∧ Nat.Coprime b c ∧ Nat.Coprime a c
+
+/--
+**Erdős Problem #123**
+
+Let $a, b, c$ be three integers which are pairwise coprime. Is every large integer
+the sum of distinct integers of the form $a^k b^l c^m$ ($k, l, m ≥ 0$), none of which
+divide any other?
+
+Equivalently: is the set $\{a^k b^l c^m : k, l, m \geq 0\}$ d-complete?
+-/
+@[category research open, AMS 11]
+theorem erdos_123 (a b c : ℕ) (h_coprime : PairwiseCoprime a b c) :
+    IsDComplete (PowersOfThree a b c) ↔ answer(sorry) := by
+  sorry
+
+/--
+Erdős and Lewin proved this conjecture when $a = 3$, $b = 5$, and $c = 7$.
+
+Reference: [ErLe96] Erdős, P. and Lewin, Mordechai,
+_$d$-complete sequences of integers_. Math. Comp. (1996), 837-840.
+-/
+@[category research solved, AMS 11]
+theorem erdos_123.variants.erdos_lewin_3_5_7 :
+    IsDComplete (PowersOfThree 3 5 7) := by
+  sorry
+
+/--
+A simpler case: the set of numbers of the form $2^k 3^l$ ($k, l ≥ 0$) is d-complete.
+
+This was initially conjectured by Erdős in 1992, who called it a "nice and difficult"
+problem, but it was quickly proven by Jansen and others using a simple inductive argument:
+- If $n = 2m$ is even, apply the inductive hypothesis to $m$ and double all summands.
+- If $n$ is odd, let $3^k$ be the largest power of $3$ with $3^k ≤ n$, and apply the
+  inductive hypothesis to $n - 3^k$ (which is even).
+
+Reference: [Er92b] Erdős, Paul, _Some of my favourite problems in various branches
+of combinatorics_. Matematiche (Catania) (1992), 231-240.
+-/
+@[category research solved, AMS 11]
+theorem erdos_123.variants.powers_2_3 :
+    IsDComplete (PowersOfTwo 2 3) := by
+  sorry
+
```

Comment:
(you're more than welcome to add the extra problem if you want to, but no pressure!)
```suggestion

-- TODO(Paul-Lez): add the last problem from the webpage. 

```